{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "015c9d61",
   "metadata": {},
   "source": [
    "# Load data and go to directory with python scripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1cc2c000",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/jendawk/Dropbox (MIT)/C Diff Recurrence Paper/Analyses/scripts\n"
     ]
    }
   ],
   "source": [
    "scripts_path = '/Users/jendawk/Dropbox (MIT)/C Diff Recurrence Paper/Analyses/scripts/'\n",
    "\n",
    "# Path to save figures\n",
    "path_to_save = '/'.join(scripts_path.split('/')[:-2]) + '/'\n",
    "\n",
    "# Path to save supplementary and figure data folders\n",
    "save_path = '/Users/jendawk/Dropbox (MIT)/C Diff Recurrence Paper/'\n",
    "%cd $scripts_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "03873a51",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/jendawk/Dropbox (MIT)/C Diff Recurrence Paper/Analyses/scripts\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/jendawk/miniconda3/envs/CodeBase/lib/python3.7/site-packages/pandas/core/generic.py:5494: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self[name] = value\n"
     ]
    }
   ],
   "source": [
    "from dataLoader import *\n",
    "from helper import *\n",
    "import scipy.stats as st\n",
    "from collections import Counter\n",
    "import sklearn\n",
    "from matplotlib import cm\n",
    "import scipy\n",
    "import  itertools\n",
    "from datetime import datetime\n",
    "\n",
    "from seaborn import clustermap\n",
    "from scipy.cluster.hierarchy import linkage\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "import os\n",
    "import time\n",
    "import pickle as pkl\n",
    "import re\n",
    "from matplotlib_venn import venn3, venn2\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "dl = dataLoader(pt_perc = {'metabs': .25, '16s': .05, 'scfa': 0, 'toxin':0}, meas_thresh = \n",
    "                {'metabs': 0, '16s': 10, 'scfa': 0, 'toxin':0}, \n",
    "                var_perc = {'metabs': 15, '16s': 5, 'scfa': 0, 'toxin':0})\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9055d62",
   "metadata": {},
   "source": [
    "# Define functions for doing over representation analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "46b8684f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import math\n",
    "def combination(n,k):\n",
    "    return math.factorial(n)/(math.factorial(k)*math.factorial(n-k))\n",
    "\n",
    "def ora(x, X):\n",
    "    N = len(X)\n",
    "    n = len(x)\n",
    "    print('n=' + str(n))\n",
    "    num_hits = {}\n",
    "    num_hits_expected = {}\n",
    "    prob = {}\n",
    "    for group in np.unique(x):\n",
    "        k = np.sum(np.array(x)==group)\n",
    "        K = np.sum(np.array(X)==group)\n",
    "        if K < 4 or k < 3 or group == 'None' or group == 'Partially Characterized Molecules':\n",
    "            print(group + ': K='+str(K) + ', k=' + str(k))\n",
    "            continue\n",
    "        num_hits[group] = k\n",
    "        num_hits_expected[group] = (K/N)*n\n",
    "        prob[group] = (combination(K,k)*combination(N-K, n-k))/combination(N,n)\n",
    "    if len(prob)==0:\n",
    "        return None\n",
    "\n",
    "    gps, pvals = zip(*prob.items())\n",
    "    reject, corrected, a1, a2 = multipletests(list(pvals), alpha=.05, method='fdr_bh')\n",
    "    fdr = {gp:corr for gp, corr in zip(gps, corrected)}\n",
    "    out_df = pd.DataFrame({'# Hits':num_hits, '# Expected Hits': num_hits_expected, 'p-value': prob, 'FDR': fdr})\n",
    "    return out_df.sort_values('p-value', ascending = True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f80b2f7",
   "metadata": {},
   "source": [
    "# Enrichment analysis for metabolites"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "02c83392",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ttest_metabs0_1_Non-recurrer.csv\n",
      "SUPER_PATHWAY\n",
      "n=228\n",
      "Energy: K=3, k=2\n",
      "Partially Characterized Molecules: K=21, k=7\n",
      "                        # Hits  # Expected Hits       p-value       FDR\n",
      "Nucleotide                  28             12.9  8.033942e-07  0.000006\n",
      "Amino Acid                  25             31.5  3.065450e-02  0.105318\n",
      "Xenobiotics                 60             65.4  4.513633e-02  0.105318\n",
      "Lipid                       89             88.8  6.462484e-02  0.113093\n",
      "Cofactors and Vitamins       6              9.0  8.185609e-02  0.114599\n",
      "Peptide                      6              8.1  1.206588e-01  0.140769\n",
      "Carbohydrate                 5              5.1  2.104611e-01  0.210461\n",
      "\n",
      "ttest_metabs0_1_Non-recurrer.csv\n",
      "SUB_PATHWAY\n",
      "n=228\n",
      "Aminosugar Metabolism: K=3, k=1\n",
      "Ceramides: K=2, k=2\n",
      "Chemical: K=14, k=1\n",
      "Dihydrosphingomyelins: K=1, k=1\n",
      "Drug - Antibiotic: K=1, k=1\n",
      "Drug - Other: K=1, k=1\n",
      "Fatty Acid Metabolism (Acyl Carnitine, Long Chain Saturated): K=3, k=1\n",
      "Fatty Acid Metabolism (Acyl Glutamine): K=3, k=1\n",
      "Glutamate Metabolism: K=6, k=1\n",
      "Glutathione Metabolism: K=2, k=2\n",
      "Glycine, Serine and Threonine Metabolism: K=4, k=1\n",
      "Hemoglobin and Porphyrin Metabolism: K=5, k=1\n",
      "Histidine Metabolism: K=7, k=2\n",
      "Inositol Metabolism: K=1, k=1\n",
      "Lactosylceramides (LCER): K=2, k=1\n",
      "Long Chain Monounsaturated Fatty Acid: K=1, k=1\n",
      "Long Chain Polyunsaturated Fatty Acid (n3 and n6): K=6, k=1\n",
      "Lysine Metabolism: K=6, k=1\n",
      "Nicotinate and Nicotinamide Metabolism: K=4, k=1\n",
      "Partially Characterized Molecules: K=21, k=7\n",
      "Phenylalanine Metabolism: K=7, k=1\n",
      "Phosphatidylcholine (PC): K=6, k=1\n",
      "Phosphatidylserine (PS): K=1, k=1\n",
      "Plasmalogen: K=2, k=2\n",
      "Polyamine Metabolism: K=6, k=2\n",
      "Primary Bile Acid Metabolism: K=10, k=1\n",
      "Progestin Steroids: K=5, k=1\n",
      "Pterin Metabolism: K=1, k=1\n",
      "Purine Metabolism, (Hypo)Xanthine/Inosine containing: K=2, k=1\n",
      "Purine Metabolism, Adenine containing: K=5, k=2\n",
      "Purine Metabolism, Guanine containing: K=5, k=2\n",
      "Pyrimidine Metabolism, Orotate containing: K=3, k=2\n",
      "Pyrimidine Metabolism, Thymine containing: K=2, k=1\n",
      "TCA Cycle: K=3, k=2\n",
      "Tocopherol Metabolism: K=7, k=2\n",
      "Tyrosine Metabolism: K=14, k=1\n",
      "Urea cycle; Arginine and Proline Metabolism: K=7, k=2\n",
      "Vitamin A Metabolism: K=4, k=1\n",
      "                                                  # Hits  # Expected Hits  \\\n",
      "Sphingomyelins                                        14              4.2   \n",
      "Dinucleotide                                          11              3.3   \n",
      "Diacylglycerol                                        16              6.6   \n",
      "Pyrimidine Metabolism, Cytidine containing             5              2.1   \n",
      "Food Component/Plant                                  48             42.9   \n",
      "Disaccharides and Oligosaccharides                     4              1.8   \n",
      "Secondary Bile Acid Metabolism                         7             10.5   \n",
      "Fatty Acid, Branched                                   3              1.2   \n",
      "Fatty Acid Metabolism (Acyl Glycine)                   5              3.0   \n",
      "Galactosyl Glycerolipids                               3              5.1   \n",
      "Leucine, Isoleucine and Valine Metabolism              3              5.1   \n",
      "Pyrimidine Metabolism, Uracil containing               4              2.4   \n",
      "Sterol                                                 4              2.4   \n",
      "Benzoate Metabolism                                    9              8.4   \n",
      "Tryptophan Metabolism                                  6              4.8   \n",
      "Fatty Acid, Dicarboxylate                              5              3.9   \n",
      "Fatty Acid, Monohydroxy                                6              5.7   \n",
      "Dipeptide                                              3              2.1   \n",
      "Lysophospholipid                                       4              4.2   \n",
      "Endocannabinoid                                        4              4.2   \n",
      "Gamma-glutamyl Amino Acid                              3              3.6   \n",
      "Methionine, Cysteine, SAM and Taurine Metabolism       3              2.4   \n",
      "Monoacylglycerol                                       3              2.7   \n",
      "\n",
      "                                                       p-value           FDR  \n",
      "Sphingomyelins                                    3.590430e-08  8.257989e-07  \n",
      "Dinucleotide                                      1.491072e-06  1.714732e-05  \n",
      "Diacylglycerol                                    2.913275e-05  2.233511e-04  \n",
      "Pyrimidine Metabolism, Cytidine containing        2.455074e-02  1.411668e-01  \n",
      "Food Component/Plant                              4.666507e-02  2.146593e-01  \n",
      "Disaccharides and Oligosaccharides                5.902825e-02  2.162012e-01  \n",
      "Secondary Bile Acid Metabolism                    6.623753e-02  2.162012e-01  \n",
      "Fatty Acid, Branched                              7.520043e-02  2.162012e-01  \n",
      "Fatty Acid Metabolism (Acyl Glycine)              1.025554e-01  2.406669e-01  \n",
      "Galactosyl Glycerolipids                          1.238428e-01  2.406669e-01  \n",
      "Leucine, Isoleucine and Valine Metabolism         1.238428e-01  2.406669e-01  \n",
      "Pyrimidine Metabolism, Uracil containing          1.360291e-01  2.406669e-01  \n",
      "Sterol                                            1.360291e-01  2.406669e-01  \n",
      "Benzoate Metabolism                               1.578383e-01  2.547720e-01  \n",
      "Tryptophan Metabolism                             1.661557e-01  2.547720e-01  \n",
      "Fatty Acid, Dicarboxylate                         1.814043e-01  2.607687e-01  \n",
      "Fatty Acid, Monohydroxy                           1.940970e-01  2.626018e-01  \n",
      "Dipeptide                                         2.276221e-01  2.641208e-01  \n",
      "Lysophospholipid                                  2.310827e-01  2.641208e-01  \n",
      "Endocannabinoid                                   2.310827e-01  2.641208e-01  \n",
      "Gamma-glutamyl Amino Acid                         2.411537e-01  2.641208e-01  \n",
      "Methionine, Cysteine, SAM and Taurine Metabolism  2.553720e-01  2.669798e-01  \n",
      "Monoacylglycerol                                  2.684462e-01  2.684462e-01  \n",
      "\n",
      "ttest_metabs1_2_Recurrer.csv\n",
      "SUPER_PATHWAY\n",
      "n=0\n",
      "None\n",
      "\n",
      "ttest_metabs1_2_Recurrer.csv\n",
      "SUB_PATHWAY\n",
      "n=0\n",
      "None\n",
      "\n",
      "ttest_metabs[1, 1.5, 2].csv\n",
      "SUPER_PATHWAY\n",
      "n=0\n",
      "None\n",
      "\n",
      "ttest_metabs[1, 1.5, 2].csv\n",
      "SUB_PATHWAY\n",
      "n=0\n",
      "None\n",
      "\n",
      "ttest_metabs1_2_Non-recurrer.csv\n",
      "SUPER_PATHWAY\n",
      "n=121\n",
      "Energy: K=3, k=1\n",
      "Partially Characterized Molecules: K=21, k=9\n",
      "Peptide: K=27, k=2\n",
      "                        # Hits  # Expected Hits   p-value       FDR\n",
      "Lipid                       43        47.126316  0.057624  0.117657\n",
      "Amino Acid                  13        16.717105  0.068237  0.117657\n",
      "Carbohydrate                 5         2.706579  0.078427  0.117657\n",
      "Xenobiotics                 33        34.707895  0.082288  0.117657\n",
      "Cofactors and Vitamins       7         4.776316  0.098047  0.117657\n",
      "Nucleotide                   8         6.846053  0.142108  0.142108\n",
      "\n",
      "ttest_metabs1_2_Non-recurrer.csv\n",
      "SUB_PATHWAY\n",
      "n=121\n",
      "Alanine and Aspartate Metabolism: K=3, k=1\n",
      "Aminosugar Metabolism: K=3, k=1\n",
      "Ascorbate and Aldarate Metabolism: K=2, k=1\n",
      "Dinucleotide: K=11, k=1\n",
      "Drug - Analgesics, Anesthetics: K=14, k=1\n",
      "Drug - Antibiotic: K=1, k=1\n",
      "Endocannabinoid: K=14, k=2\n",
      "Fatty Acid Metabolism (Acyl Carnitine, Monounsaturated): K=5, k=2\n",
      "Fatty Acid Metabolism (Acyl Glutamine): K=3, k=1\n",
      "Fructose, Mannose and Galactose Metabolism: K=2, k=1\n",
      "Gamma-glutamyl Amino Acid: K=12, k=2\n",
      "Glutamate Metabolism: K=6, k=1\n",
      "Glycine, Serine and Threonine Metabolism: K=4, k=1\n",
      "Hemoglobin and Porphyrin Metabolism: K=5, k=2\n",
      "Hexosylceramides (HCER): K=3, k=1\n",
      "Histidine Metabolism: K=7, k=1\n",
      "Inositol Metabolism: K=1, k=1\n",
      "Lactosylceramides (LCER): K=2, k=1\n",
      "Long Chain Polyunsaturated Fatty Acid (n3 and n6): K=6, k=2\n",
      "Lysoplasmalogen: K=2, k=1\n",
      "Methionine, Cysteine, SAM and Taurine Metabolism: K=8, k=1\n",
      "Pantothenate and CoA Metabolism: K=1, k=1\n",
      "Partially Characterized Molecules: K=21, k=9\n",
      "Polyamine Metabolism: K=6, k=1\n",
      "Progestin Steroids: K=5, k=2\n",
      "Purine Metabolism, Adenine containing: K=5, k=1\n",
      "Purine Metabolism, Guanine containing: K=5, k=1\n",
      "Pyrimidine Metabolism, Cytidine containing: K=7, k=1\n",
      "Pyrimidine Metabolism, Orotate containing: K=3, k=1\n",
      "Pyrimidine Metabolism, Thymine containing: K=2, k=1\n",
      "Pyrimidine Metabolism, Uracil containing: K=8, k=2\n",
      "Sphingosines: K=2, k=1\n",
      "Sterol: K=8, k=1\n",
      "TCA Cycle: K=3, k=1\n",
      "Tetrahydrobiopterin Metabolism: K=1, k=1\n",
      "Thiamine Metabolism: K=2, k=1\n",
      "Tocopherol Metabolism: K=7, k=1\n",
      "Tryptophan Metabolism: K=16, k=2\n",
      "Urea cycle; Arginine and Proline Metabolism: K=7, k=2\n",
      "                                                 # Hits  # Expected Hits  \\\n",
      "Secondary Bile Acid Metabolism                       11         5.572368   \n",
      "Primary Bile Acid Metabolism                          5         1.592105   \n",
      "Fatty Acid Metabolism (Acyl Carnitine, Hydroxy)       3         0.636842   \n",
      "Corticosteroids                                       3         0.796053   \n",
      "Food Component/Plant                                 28        22.767105   \n",
      "Disaccharides and Oligosaccharides                    3         0.955263   \n",
      "Benzoate Metabolism                                   3         4.457895   \n",
      "Androgenic Steroids                                   3         2.547368   \n",
      "Leucine, Isoleucine and Valine Metabolism             3         2.706579   \n",
      "Fatty Acid, Monohydroxy                               3         3.025000   \n",
      "\n",
      "                                                  p-value       FDR  \n",
      "Secondary Bile Acid Metabolism                   0.009694  0.044477  \n",
      "Primary Bile Acid Metabolism                     0.010408  0.044477  \n",
      "Fatty Acid Metabolism (Acyl Carnitine, Hydroxy)  0.013343  0.044477  \n",
      "Corticosteroids                                  0.028151  0.070377  \n",
      "Food Component/Plant                             0.040901  0.079171  \n",
      "Disaccharides and Oligosaccharides               0.047502  0.079171  \n",
      "Benzoate Metabolism                              0.173662  0.246994  \n",
      "Androgenic Steroids                              0.239816  0.246994  \n",
      "Leucine, Isoleucine and Valine Metabolism        0.245019  0.246994  \n",
      "Fatty Acid, Monohydroxy                          0.246994  0.246994  \n",
      "\n",
      "ttest_metabs2.csv\n",
      "SUPER_PATHWAY\n",
      "n=23\n",
      "Amino Acid: K=105, k=1\n",
      "Carbohydrate: K=17, k=1\n",
      "Cofactors and Vitamins: K=30, k=1\n",
      "             # Hits  # Expected Hits   p-value       FDR\n",
      "Lipid            15         8.957895  0.006256  0.012513\n",
      "Xenobiotics       5         6.597368  0.149470  0.149470\n",
      "\n",
      "ttest_metabs2.csv\n",
      "SUB_PATHWAY\n",
      "n=23\n",
      "Androgenic Steroids: K=16, k=2\n",
      "Corticosteroids: K=5, k=1\n",
      "Disaccharides and Oligosaccharides: K=6, k=1\n",
      "Drug - Analgesics, Anesthetics: K=14, k=1\n",
      "Endocannabinoid: K=14, k=2\n",
      "Estrogenic Steroids: K=2, k=1\n",
      "Hemoglobin and Porphyrin Metabolism: K=5, k=1\n",
      "Lysophospholipid: K=14, k=1\n",
      "Phosphatidylcholine (PC): K=6, k=1\n",
      "Progestin Steroids: K=5, k=1\n",
      "Secondary Bile Acid Metabolism: K=35, k=2\n",
      "Urea cycle; Arginine and Proline Metabolism: K=7, k=1\n",
      "                      # Hits  # Expected Hits   p-value       FDR\n",
      "Sphingomyelins             4         0.423684  0.000498  0.000995\n",
      "Food Component/Plant       4         4.327632  0.214439  0.214439\n",
      "\n",
      "ttest_metabs0_1_Recurrer.csv\n",
      "SUPER_PATHWAY\n",
      "n=43\n",
      "Cofactors and Vitamins: K=30, k=1\n",
      "             # Hits  # Expected Hits   p-value       FDR\n",
      "Xenobiotics       5        12.334211  0.004186  0.020930\n",
      "Lipid            24        16.747368  0.008928  0.022321\n",
      "Nucleotide        6         2.432895  0.020999  0.034998\n",
      "Peptide           3         1.527632  0.131836  0.136178\n",
      "Amino Acid        4         5.940789  0.136178  0.136178\n",
      "\n",
      "ttest_metabs0_1_Recurrer.csv\n",
      "SUB_PATHWAY\n",
      "n=43\n",
      "Acetylated Peptides: K=7, k=1\n",
      "Benzoate Metabolism: K=28, k=1\n",
      "Dinucleotide: K=11, k=2\n",
      "Dipeptide: K=7, k=1\n",
      "Fatty Acid Metabolism (Acyl Carnitine, Hydroxy): K=4, k=2\n",
      "Fatty Acid Metabolism (Acyl Carnitine, Monounsaturated): K=5, k=1\n",
      "Fatty Acid, Dicarboxylate: K=13, k=2\n",
      "Fatty Acid, Monohydroxy: K=19, k=2\n",
      "Gamma-glutamyl Amino Acid: K=12, k=1\n",
      "Hemoglobin and Porphyrin Metabolism: K=5, k=1\n",
      "Histidine Metabolism: K=7, k=1\n",
      "Long Chain Monounsaturated Fatty Acid: K=1, k=1\n",
      "Long Chain Polyunsaturated Fatty Acid (n3 and n6): K=6, k=1\n",
      "Lysophospholipid: K=14, k=2\n",
      "Monoacylglycerol: K=9, k=1\n",
      "Plasmalogen: K=2, k=2\n",
      "Purine Metabolism, (Hypo)Xanthine/Inosine containing: K=2, k=1\n",
      "Pyrimidine Metabolism, Cytidine containing: K=7, k=2\n",
      "Pyrimidine Metabolism, Uracil containing: K=8, k=1\n",
      "Sterol: K=8, k=1\n",
      "                       # Hits  # Expected Hits   p-value       FDR\n",
      "Sphingomyelins              5         0.792105  0.000579  0.002317\n",
      "Diacylglycerol              4         1.244737  0.024949  0.046552\n",
      "Food Component/Plant        4         8.090789  0.043012  0.046552\n",
      "Tryptophan Metabolism       3         0.905263  0.046552  0.046552\n",
      "\n",
      "ttest_metabs0.csv\n",
      "SUPER_PATHWAY\n",
      "n=0\n",
      "None\n",
      "\n",
      "ttest_metabs0.csv\n",
      "SUB_PATHWAY\n",
      "n=0\n",
      "None\n",
      "\n",
      "ttest_metabs1.csv\n",
      "SUPER_PATHWAY\n",
      "n=0\n",
      "None\n",
      "\n",
      "ttest_metabs1.csv\n",
      "SUB_PATHWAY\n",
      "n=0\n",
      "None\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set path to where univariate analysis results of metabolites is stored\n",
    "path = '/Users/jendawk/Dropbox (MIT)/C Diff Recurrence Paper/Analyses/univariate_analysis/metabs/'\n",
    "for file in os.listdir(path):\n",
    "    if '.csv' not in file:\n",
    "        continue\n",
    "    if 'ranksum' in file:\n",
    "        continue\n",
    "        \n",
    "    df = pd.read_csv(path + file, index_col = 0)\n",
    "    mets = df.index.values[df['BH corrected']<= 0.05]\n",
    "    \n",
    "    for pathway in ['SUPER_PATHWAY','SUB_PATHWAY']:\n",
    "        K = dl.col_mat_mets[pathway].loc[df.index.values]\n",
    "        k = dl.col_mat_mets[pathway].loc[mets]\n",
    "        print(file)\n",
    "        print(pathway)\n",
    "        print(ora(k,K))\n",
    "        print('')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "df491ead",
   "metadata": {},
   "source": [
    "# Enrichment analysis for ASVs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "b78bb813",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "deseq2_16sNon-recurrer1_2.csv\n",
      "rdp\n",
      "Family\n",
      "n=12\n",
      "Bacteroidaceae: K=14, k=1\n",
      "Erysipelotrichaceae: K=12, k=1\n",
      "Prevotellaceae: K=10, k=1\n",
      "Ruminococcaceae: K=18, k=1\n",
      "Veillonellaceae: K=21, k=1\n",
      "Verrucomicrobiaceae: K=2, k=1\n",
      "                 # Hits  # Expected Hits   p-value       FDR\n",
      "Lachnospiraceae       6          3.06383  0.041147  0.041147\n",
      "Genus\n",
      "n=12\n",
      "Akkermansia: K=2, k=1\n",
      "Bacteroides: K=13, k=1\n",
      "Clostridium_XVIII: K=3, k=1\n",
      "Faecalibacterium: K=5, k=1\n",
      "Fusicatenibacter: K=1, k=1\n",
      "Prevotella: K=10, k=1\n",
      "Roseburia: K=4, k=1\n",
      "Ruminococcus2: K=3, k=1\n",
      "Veillonella: K=9, k=1\n",
      "                  # Hits  # Expected Hits   p-value       FDR\n",
      "Clostridium_XlVa       3         1.023697  0.058486  0.058486\n",
      "\n",
      "deseq2_16sNon-recurrer1_2.csv\n",
      "silva\n",
      "Family\n",
      "n=12\n",
      "Akkermansiaceae: K=2, k=1\n",
      "Bacteroidaceae: K=13, k=1\n",
      "Erysipelatoclostridiaceae: K=5, k=1\n",
      "Prevotellaceae: K=10, k=1\n",
      "Ruminococcaceae: K=13, k=1\n",
      "Veillonellaceae: K=19, k=1\n",
      "                 # Hits  # Expected Hits   p-value       FDR\n",
      "Lachnospiraceae       6         3.128205  0.044841  0.044841\n",
      "Genus\n",
      "n=11\n",
      "Akkermansia: K=2, k=1\n",
      "Bacteroides: K=13, k=1\n",
      "Erysipelatoclostridium: K=3, k=1\n",
      "Faecalibacterium: K=4, k=1\n",
      "Fusicatenibacter: K=1, k=1\n",
      "Lachnospiraceae NK4A136 group: K=4, k=1\n",
      "Lachnospiraceae UCG-009: K=1, k=1\n",
      "Prevotella: K=7, k=1\n",
      "Roseburia: K=2, k=1\n",
      "Veillonella: K=9, k=1\n",
      "[Ruminococcus] torques group: K=4, k=1\n",
      "None\n",
      "\n",
      "deseq2_16s0.csv\n",
      "rdp\n",
      "Family\n",
      "n=10\n",
      "Bacteroidaceae: K=14, k=1\n",
      "Methanobacteriaceae: K=1, k=1\n",
      "Porphyromonadaceae: K=4, k=1\n",
      "Ruminococcaceae: K=18, k=2\n",
      "Veillonellaceae: K=21, k=1\n",
      "                 # Hits  # Expected Hits   p-value       FDR\n",
      "Lachnospiraceae       4         2.553191  0.153024  0.153024\n",
      "Genus\n",
      "n=8\n",
      "Anaerostipes: K=2, k=1\n",
      "Bacteroides: K=13, k=1\n",
      "Blautia: K=9, k=1\n",
      "Eisenbergiella: K=2, k=1\n",
      "Faecalibacterium: K=5, k=1\n",
      "Methanobrevibacter: K=1, k=1\n",
      "Odoribacter: K=1, k=1\n",
      "Veillonella: K=9, k=1\n",
      "None\n",
      "\n",
      "deseq2_16s0.csv\n",
      "silva\n",
      "Family\n",
      "n=10\n",
      "Bacteroidaceae: K=13, k=1\n",
      "Marinifilaceae: K=1, k=1\n",
      "Methanobacteriaceae: K=1, k=1\n",
      "Monoglobaceae: K=1, k=1\n",
      "Ruminococcaceae: K=13, k=1\n",
      "Veillonellaceae: K=19, k=1\n",
      "                 # Hits  # Expected Hits   p-value       FDR\n",
      "Lachnospiraceae       4         2.606838  0.159481  0.159481\n",
      "Genus\n",
      "n=10\n",
      "Anaerostipes: K=2, k=1\n",
      "Bacteroides: K=13, k=1\n",
      "Blautia: K=9, k=1\n",
      "Eisenbergiella: K=2, k=1\n",
      "Faecalibacterium: K=4, k=1\n",
      "Methanobrevibacter: K=1, k=1\n",
      "Monoglobus: K=1, k=1\n",
      "Odoribacter: K=1, k=1\n",
      "Veillonella: K=9, k=1\n",
      "[Eubacterium] hallii group: K=3, k=1\n",
      "None\n",
      "\n",
      "deseq2_16sRecurrer1_2.csv\n",
      "rdp\n",
      "Family\n",
      "n=8\n",
      "Bacteroidaceae: K=14, k=1\n",
      "Clostridiaceae_1: K=12, k=2\n",
      "Desulfovibrionaceae: K=1, k=1\n",
      "Enterococcaceae: K=4, k=1\n",
      "Lachnospiraceae: K=60, k=1\n",
      "Veillonellaceae: K=21, k=1\n",
      "Verrucomicrobiaceae: K=2, k=1\n",
      "None\n",
      "Genus\n",
      "n=8\n",
      "Akkermansia: K=2, k=1\n",
      "Bacteroides: K=13, k=1\n",
      "Bilophila: K=1, k=1\n",
      "Clostridium_sensu_stricto: K=12, k=2\n",
      "Enterococcus: K=3, k=1\n",
      "Hungatella: K=1, k=1\n",
      "Propionispira: K=1, k=1\n",
      "None\n",
      "\n",
      "deseq2_16sRecurrer1_2.csv\n",
      "silva\n",
      "Family\n",
      "n=8\n",
      "Akkermansiaceae: K=2, k=1\n",
      "Bacteroidaceae: K=13, k=1\n",
      "Clostridiaceae: K=12, k=2\n",
      "Desulfovibrionaceae: K=1, k=1\n",
      "Enterococcaceae: K=3, k=1\n",
      "Lachnospiraceae: K=61, k=1\n",
      "Selenomonadaceae: K=2, k=1\n",
      "None\n",
      "Genus\n",
      "n=7\n",
      "Akkermansia: K=2, k=1\n",
      "Bacteroides: K=13, k=1\n",
      "Bilophila: K=1, k=1\n",
      "Clostridium sensu stricto 1: K=9, k=1\n",
      "Clostridium sensu stricto 2: K=1, k=1\n",
      "Enterococcus: K=3, k=1\n",
      "Hungatella: K=2, k=1\n",
      "None\n",
      "\n",
      "deseq2_16s1.csv\n",
      "rdp\n",
      "Family\n",
      "n=8\n",
      "Acidaminococcaceae: K=2, k=1\n",
      "Bacteroidaceae: K=14, k=1\n",
      "Bifidobacteriaceae: K=5, k=1\n",
      "Enterobacteriaceae: K=14, k=1\n",
      "Lachnospiraceae: K=60, k=1\n",
      "Ruminococcaceae: K=18, k=1\n",
      "Streptococcaceae: K=7, k=1\n",
      "Veillonellaceae: K=21, k=1\n",
      "None\n",
      "Genus\n",
      "n=8\n",
      "Bacteroides: K=13, k=1\n",
      "Bifidobacterium: K=4, k=1\n",
      "Blautia: K=9, k=1\n",
      "Faecalibacterium: K=5, k=1\n",
      "Phascolarctobacterium: K=1, k=1\n",
      "Streptococcus: K=6, k=1\n",
      "Veillonella: K=9, k=1\n",
      "Yersinia: K=1, k=1\n",
      "None\n",
      "\n",
      "deseq2_16s1.csv\n",
      "silva\n",
      "Family\n",
      "n=7\n",
      "Acidaminococcaceae: K=2, k=1\n",
      "Bacteroidaceae: K=13, k=1\n",
      "Bifidobacteriaceae: K=5, k=1\n",
      "Lachnospiraceae: K=61, k=1\n",
      "Ruminococcaceae: K=13, k=1\n",
      "Streptococcaceae: K=7, k=1\n",
      "Veillonellaceae: K=19, k=1\n",
      "None\n",
      "Genus\n",
      "n=7\n",
      "Bacteroides: K=13, k=1\n",
      "Bifidobacterium: K=4, k=1\n",
      "Blautia: K=9, k=1\n",
      "Faecalibacterium: K=4, k=1\n",
      "Phascolarctobacterium: K=1, k=1\n",
      "Streptococcus: K=6, k=1\n",
      "Veillonella: K=9, k=1\n",
      "None\n",
      "\n",
      "deseq2_16s2.csv\n",
      "rdp\n",
      "Family\n",
      "n=29\n",
      "Acidaminococcaceae: K=2, k=1\n",
      "Bifidobacteriaceae: K=5, k=1\n",
      "Desulfovibrionaceae: K=1, k=1\n",
      "Enterobacteriaceae: K=14, k=2\n",
      "Erysipelotrichaceae: K=12, k=1\n",
      "Veillonellaceae: K=21, k=1\n",
      "Verrucomicrobiaceae: K=2, k=1\n",
      "                 # Hits  # Expected Hits   p-value       FDR\n",
      "Ruminococcaceae       6         2.221277  0.010918  0.032754\n",
      "Lachnospiraceae      11         7.404255  0.047690  0.059590\n",
      "Bacteroidaceae        4         1.727660  0.059590  0.059590\n",
      "Genus\n",
      "n=28\n",
      "Akkermansia: K=2, k=1\n",
      "Anaerostipes: K=2, k=1\n",
      "Anaerotruncus: K=1, k=1\n",
      "Bifidobacterium: K=4, k=1\n",
      "Bilophila: K=1, k=1\n",
      "Citrobacter: K=1, k=1\n",
      "Clostridium_IV: K=3, k=2\n",
      "Clostridium_XlVa: K=18, k=2\n",
      "Coprobacillus: K=1, k=1\n",
      "Faecalibacterium: K=5, k=1\n",
      "Fusicatenibacter: K=1, k=1\n",
      "Oscillibacter: K=1, k=1\n",
      "Phascolarctobacterium: K=1, k=1\n",
      "Proteus: K=2, k=1\n",
      "Roseburia: K=4, k=2\n",
      "Subdoligranulum: K=1, k=1\n",
      "Veillonella: K=9, k=1\n",
      "             # Hits  # Expected Hits   p-value       FDR\n",
      "Blautia           4         1.194313  0.017247  0.034493\n",
      "Bacteroides       4         1.725118  0.058770  0.058770\n",
      "\n",
      "deseq2_16s2.csv\n",
      "silva\n",
      "Family\n",
      "n=29\n",
      "Acidaminococcaceae: K=2, k=1\n",
      "Akkermansiaceae: K=2, k=1\n",
      "Bifidobacteriaceae: K=5, k=1\n",
      "Desulfovibrionaceae: K=1, k=1\n",
      "Enterobacteriaceae: K=9, k=1\n",
      "Erysipelatoclostridiaceae: K=5, k=1\n",
      "Morganellaceae: K=3, k=1\n",
      "Oscillospiraceae: K=2, k=1\n",
      "Veillonellaceae: K=19, k=1\n",
      "[Eubacterium] coprostanoligenes group: K=1, k=1\n",
      "                 # Hits  # Expected Hits   p-value       FDR\n",
      "Bacteroidaceae        4         1.611111  0.048557  0.053386\n",
      "Ruminococcaceae       4         1.611111  0.048557  0.053386\n",
      "Lachnospiraceae      11         7.559829  0.053386  0.053386\n",
      "Genus\n",
      "n=27\n",
      "Agathobacter: K=2, k=1\n",
      "Akkermansia: K=2, k=1\n",
      "Anaerostipes: K=2, k=1\n",
      "Bifidobacterium: K=4, k=1\n",
      "Bilophila: K=1, k=1\n",
      "Coprobacillus: K=1, k=1\n",
      "Faecalibacterium: K=4, k=1\n",
      "Fusicatenibacter: K=1, k=1\n",
      "Incertae Sedis: K=4, k=2\n",
      "Lachnoclostridium: K=7, k=1\n",
      "Oscillibacter: K=1, k=1\n",
      "Phascolarctobacterium: K=1, k=1\n",
      "Pluralibacter: K=1, k=1\n",
      "Proteus: K=2, k=1\n",
      "Roseburia: K=2, k=1\n",
      "UBA1819: K=1, k=1\n",
      "Veillonella: K=9, k=1\n",
      "[Ruminococcus] gnavus group: K=1, k=1\n",
      "             # Hits  # Expected Hits   p-value       FDR\n",
      "Blautia           4         1.070485  0.011848  0.023696\n",
      "Bacteroides       4         1.546256  0.042901  0.042901\n",
      "\n",
      "deseq2_16sNon-recurrer0_1.csv\n",
      "rdp\n",
      "Family\n",
      "n=31\n",
      "Erysipelotrichaceae: K=12, k=2\n",
      "Fusobacteriaceae: K=9, k=1\n",
      "Peptostreptococcaceae: K=6, k=1\n",
      "Prevotellaceae: K=10, k=1\n",
      "Rikenellaceae: K=4, k=1\n",
      "Ruminococcaceae: K=18, k=2\n",
      "Veillonellaceae: K=21, k=2\n",
      "Verrucomicrobiaceae: K=2, k=1\n",
      "                    # Hits  # Expected Hits   p-value       FDR\n",
      "Lactobacillaceae         5         1.714894  0.014407  0.057626\n",
      "Bacteroidaceae           4         1.846809  0.071448  0.095264\n",
      "Enterobacteriaceae       4         1.846809  0.071448  0.095264\n",
      "Lachnospiraceae          7         7.914894  0.166053  0.166053\n",
      "Genus\n",
      "n=28\n",
      "Akkermansia: K=2, k=1\n",
      "Alistipes: K=4, k=1\n",
      "Anaerostipes: K=2, k=1\n",
      "Clostridium_XI: K=1, k=1\n",
      "Clostridium_XVIII: K=3, k=2\n",
      "Clostridium_XlVa: K=18, k=1\n",
      "Coprococcus: K=1, k=1\n",
      "Fusicatenibacter: K=1, k=1\n",
      "Fusobacterium: K=9, k=1\n",
      "Megasphaera: K=1, k=1\n",
      "Morganella: K=1, k=1\n",
      "Oscillibacter: K=1, k=1\n",
      "Pediococcus: K=2, k=1\n",
      "Prevotella: K=10, k=1\n",
      "Proteus: K=2, k=1\n",
      "Roseburia: K=4, k=1\n",
      "Ruminococcus2: K=3, k=1\n",
      "Veillonella: K=9, k=1\n",
      "Yersinia: K=1, k=1\n",
      "               # Hits  # Expected Hits  p-value      FDR\n",
      "Lactobacillus       4         1.459716  0.03505  0.05877\n",
      "Bacteroides         4         1.725118  0.05877  0.05877\n",
      "\n",
      "deseq2_16sNon-recurrer0_1.csv\n",
      "silva\n",
      "Family\n",
      "n=30\n",
      "Akkermansiaceae: K=2, k=1\n",
      "Enterobacteriaceae: K=9, k=1\n",
      "Erysipelatoclostridiaceae: K=5, k=2\n",
      "Fusobacteriaceae: K=9, k=1\n",
      "Monoglobaceae: K=1, k=1\n",
      "Morganellaceae: K=3, k=2\n",
      "Oscillospiraceae: K=2, k=1\n",
      "Peptostreptococcaceae: K=6, k=1\n",
      "Prevotellaceae: K=10, k=1\n",
      "Rikenellaceae: K=4, k=1\n",
      "Veillonellaceae: K=19, k=2\n",
      "                  # Hits  # Expected Hits   p-value       FDR\n",
      "Lactobacillaceae       5         1.666667  0.012791  0.038372\n",
      "Bacteroidaceae         4         1.666667  0.053567  0.080351\n",
      "Lachnospiraceae        7         7.820513  0.169234  0.169234\n",
      "Genus\n",
      "n=28\n",
      "Akkermansia: K=2, k=1\n",
      "Alistipes: K=4, k=1\n",
      "Anaerostipes: K=2, k=1\n",
      "Clostridioides: K=1, k=1\n",
      "Coprococcus: K=1, k=1\n",
      "Erysipelatoclostridium: K=3, k=1\n",
      "Erysipelotrichaceae UCG-003: K=1, k=1\n",
      "Fusicatenibacter: K=1, k=1\n",
      "Fusobacterium: K=9, k=1\n",
      "Megasphaera: K=1, k=1\n",
      "Monoglobus: K=1, k=1\n",
      "Morganella: K=1, k=1\n",
      "Oscillibacter: K=1, k=1\n",
      "Pediococcus: K=2, k=1\n",
      "Prevotella: K=7, k=1\n",
      "Proteus: K=2, k=1\n",
      "Roseburia: K=2, k=1\n",
      "Veillonella: K=9, k=1\n",
      "[Eubacterium] hallii group: K=3, k=1\n",
      "[Ruminococcus] torques group: K=4, k=1\n",
      "               # Hits  # Expected Hits   p-value       FDR\n",
      "Lactobacillus       4         1.356828  0.027937  0.047799\n",
      "Bacteroides         4         1.603524  0.047799  0.047799\n",
      "\n",
      "deseq2_16sRecurrer0_1.csv\n",
      "rdp\n",
      "Family\n",
      "n=21\n",
      "Bacteroidaceae: K=14, k=2\n",
      "Desulfovibrionaceae: K=1, k=1\n",
      "Enterobacteriaceae: K=14, k=1\n",
      "Enterococcaceae: K=4, k=1\n",
      "Porphyromonadaceae: K=4, k=2\n",
      "Prevotellaceae: K=10, k=1\n",
      "Rikenellaceae: K=4, k=1\n",
      "Streptococcaceae: K=7, k=1\n",
      "Verrucomicrobiaceae: K=2, k=1\n",
      "                  # Hits  # Expected Hits   p-value       FDR\n",
      "Lactobacillaceae       4         1.161702  0.017146  0.051437\n",
      "Lachnospiraceae        3         5.361702  0.106026  0.159039\n",
      "Veillonellaceae        3         1.876596  0.182577  0.182577\n",
      "Genus\n",
      "n=20\n",
      "Akkermansia: K=2, k=1\n",
      "Alistipes: K=4, k=1\n",
      "Bacteroides: K=13, k=2\n",
      "Bilophila: K=1, k=1\n",
      "Dialister: K=3, k=1\n",
      "Dorea: K=1, k=1\n",
      "Enterobacter: K=1, k=1\n",
      "Enterococcus: K=3, k=1\n",
      "Parabacteroides: K=3, k=2\n",
      "Pediococcus: K=2, k=1\n",
      "Prevotella: K=10, k=1\n",
      "Propionispira: K=1, k=1\n",
      "Ruminococcus2: K=3, k=1\n",
      "Streptococcus: K=6, k=1\n",
      "Veillonella: K=9, k=1\n",
      "               # Hits  # Expected Hits   p-value       FDR\n",
      "Lactobacillus       3         1.042654  0.060858  0.060858\n",
      "\n",
      "deseq2_16sRecurrer0_1.csv\n",
      "silva\n",
      "Family\n",
      "n=21\n",
      "Akkermansiaceae: K=2, k=1\n",
      "Bacteroidaceae: K=13, k=2\n",
      "Desulfovibrionaceae: K=1, k=1\n",
      "Enterobacteriaceae: K=9, k=1\n",
      "Enterococcaceae: K=3, k=1\n",
      "Prevotellaceae: K=10, k=1\n",
      "Rikenellaceae: K=4, k=1\n",
      "Selenomonadaceae: K=2, k=1\n",
      "Streptococcaceae: K=7, k=1\n",
      "Tannerellaceae: K=3, k=2\n",
      "Veillonellaceae: K=19, k=2\n",
      "                  # Hits  # Expected Hits   p-value       FDR\n",
      "Lactobacillaceae       4         1.166667  0.017386  0.034773\n",
      "Lachnospiraceae        3         5.474359  0.098493  0.098493\n",
      "Genus\n",
      "n=20\n",
      "Akkermansia: K=2, k=1\n",
      "Alistipes: K=4, k=1\n",
      "Bacteroides: K=13, k=2\n",
      "Bilophila: K=1, k=1\n",
      "Dialister: K=2, k=1\n",
      "Dorea: K=1, k=1\n",
      "Enterococcus: K=3, k=1\n",
      "Klebsiella: K=2, k=1\n",
      "Parabacteroides: K=3, k=2\n",
      "Pediococcus: K=2, k=1\n",
      "Prevotella: K=7, k=1\n",
      "Sellimonas: K=1, k=1\n",
      "Streptococcus: K=6, k=1\n",
      "Veillonella: K=9, k=1\n",
      "[Ruminococcus] torques group: K=4, k=1\n",
      "               # Hits  # Expected Hits   p-value       FDR\n",
      "Lactobacillus       3         0.969163  0.051456  0.051456\n",
      "\n"
     ]
    }
   ],
   "source": [
    "path = '/Users/jendawk/Dropbox (MIT)/C Diff Recurrence Paper/Analyses/univariate_analysis/16s/'\n",
    "path_taxa = '/Users/jendawk/Dropbox (MIT)/C Diff Recurrence Paper/Analyses/scripts/inputs/'\n",
    "taxa_rdp = pd.read_csv(path_taxa + 'dada2-taxonomy-rdp.csv', index_col = 0)\n",
    "taxa_silva = pd.read_csv(path_taxa + 'dada2-taxonomy-silva.csv', index_col = 0)\n",
    "for file in os.listdir(path):\n",
    "    if '.csv' not in file:\n",
    "        continue\n",
    "        \n",
    "    df = pd.read_csv(path + file, index_col = 0)\n",
    "    for fname, taxa in zip(['rdp','silva'],[taxa_rdp, taxa_silva]):\n",
    "        FAM = taxa[df.index.values].loc['Family'].dropna()\n",
    "        GEN = taxa[df.index.values].loc['Genus'].dropna()\n",
    "\n",
    "        mets = df.index.values[df['padj']<= 0.05]\n",
    "        fam = taxa[mets].loc['Family'].dropna()\n",
    "        gen = taxa[mets].loc['Genus'].dropna()\n",
    "        print(file)\n",
    "        print(fname)\n",
    "        print('Family')\n",
    "        print(ora(fam, FAM))\n",
    "        print('Genus')\n",
    "        print(ora(gen,GEN))\n",
    "        print('')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (CodeBase)",
   "language": "python",
   "name": "codebase"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
